### DISCUSSION TITLE
How to run MAUI on Windows from command line?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#7490] [Q&A]
I know one can do:

```
dotnet build -t:Run -f net6.0-android
dotnet build -t:Run -f net6.0-ios
dotnet build -t:Run -f net6.0-maccatalyst
```

but that does not work for Windows as one would expect:

```
dotnet build -t:Run -f net6.0-windows10.0.19041.0  # fails
```

Can I run my MAUI app for Windows from command line easily? Or is it not supported at the moment?

Thanks!

**Comment by Ombrelin** [comment]
Not working for me either, getting the following : 

```
PS C:\Users\User\git\plex-rich-presence\src\PlexRichPresence.UI> dotnet build -t:Run -f net6.0-windows10.0.19041.0
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Users\User\git\plex-rich-presence\src\PlexRichPresence.UI\PlexRichPresence.UI.csproj (in 762 ms).
  Unhandled exception. System.DllNotFoundException: Unable to load DLL 'Microsoft.ui.xaml.dll' or one of its dependencies: The specif
  ied module could not be found. (0x8007007E)
     at PlexRichPresence.UI.WinUI.Program.XamlCheckProcessRequirements()
     at PlexRichPresence.UI.WinUI.Program.Main(String[] args) in C:\Users\User\git\plex-rich-presence\src\PlexRichPresence.UI\obj\Deb
  ug\net6.0-windows10.0.19041.0\win10-x64\Platforms\Windows\App.g.i.cs:line 28
C:\Program Files\dotnet\sdk\6.0.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(901,5): error MSB3073: The command "C:\U
sers\User\git\plex-rich-presence\src\PlexRichPresence.UI\bin\Debug\net6.0-windows10.0.19041.0\win10-x64\PlexRichPresence.UI.exe " exi
ted with code -532462766. [C:\Users\User\git\plex-rich-presence\src\PlexRichPresence.UI\PlexRichPresence.UI.csproj]

Build FAILED.

C:\Program Files\dotnet\sdk\6.0.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(901,5): error MSB3073: The command "C:\U
sers\User\git\plex-rich-presence\src\PlexRichPresence.UI\bin\Debug\net6.0-windows10.0.19041.0\win10-x64\PlexRichPresence.UI.exe " exi
ted with code -532462766. [C:\Users\User\git\plex-rich-presence\src\PlexRichPresence.UI\PlexRichPresence.UI.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.88

```

**Reply by MartyIX** [reply]
imho the reason why this does not work is related to https://docs.microsoft.com/en-gb/dotnet/maui/windows/deployment/overview#current-limitations. But I might be wrong. Hopefully, it will get better soon.

**Reply by DamianSuess** [reply]
Any official word on a clean example of how to do this, other than what is outlined in the [Publish-CLI](https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-cli)

**Comment by Tsingbo-Kooboo** [comment]
Any updates? I encountered the same problem.

**Reply by MartyIX** [reply]
Based on command provided in [docs](https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-unpackaged-cli?view=net-maui-8.0#publish) (unpackaged app), one can do this in PowerShell in one line:

```powershell
dotnet publish -f net8.0-windows10.0.19041.0 -c Debug -p:PublishReadyToRun=true -p:WindowsPackageType=None && .\bin\Debug\net8.0-windows10.0.19041.0\win10-x64\publish\<YOUR_PROJECT_NAME>.exe
```
(using `command1 && command2` syntax)

It's not what you are looking for but I think it's the best what one can achieve right now.

**Comment by potocnik** [comment]
Another way that worked - in vscode click on Run -> Add Configuration and select .NET maui. You can then Run -> Start debugging.

Looks like it's doing a build (maui: Build), followed by launch command. It would be nice if all this was nicely integrated into dotnet run ...

**Comment by sdheer** [comment]
I use the following commands:
    `dotnet build -f net8.0-windows10.0.19041.0 -c Debug -p:PublishReadyToRun=true -p:WindowsPackageType=None`
    or
    `dotnet run -f net8.0-windows10.0.19041.0 -c Debug -p:PublishReadyToRun=true -p:WindowsPackageType=None`
